core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -148,10 +148,15 @@ class AdminClient(val time: Time,
       throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
-    val consumers = metadata.members.map { consumer =>
-      val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
-      ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
-    }.toList
+    val consumers = metadata.state match {
+      case "PreparingRebalance" =>
```

Comment:
Thanks. I thought about that one but wasn't sure if consumers have valid assignments in that state or not. I'll update the patch.